commands/create.go

```diff
@@ -524,17 +525,30 @@ func updateUserdataFile(driverOpts *rpcdriver.RPCFlags, machineName, hostname, u
 	if err != nil {
 		return err
 	}
-	// Remove the shebang
-	customScriptContent = regexp.MustCompile(`^#!.*\n`).ReplaceAll(customScriptContent, nil)
 
-	modifiedUserdataFile, err := ioutil.TempFile("", "modified-user-data")
+	modifiedUserdataFile, err := os.CreateTemp("", "modified-user-data")
 	if err != nil {
 		return fmt.Errorf("[updateUserdataFile] unable to create tempfile [%v]\nError returned\n[%v]", modifiedUserdataFile, err)
 	}
 	defer modifiedUserdataFile.Close()
 
-	if err := replaceUserdataFile(machineName, machineOS, hostname, userdataContent, customScriptContent, modifiedUserdataFile); err != nil {
-		return err
+	if bytes.HasPrefix(customScriptContent, []byte("#cloud-config")) || bytes.HasPrefix(customScriptContent, []byte("## template: jinja\n#cloud-config")) {
+		// The custom install script is a cloudinit file. This is specifically to address v2prov, where there is no
+		// machine-driver agnostic user-data flag.
+		if userdataFile != "" {
+			return fmt.Errorf("cannot pass both machine-driver specific user-data flag %s and generic user-data flag %s in '#cloud-config' format", userdataFlag, customInstallScriptFlag)
+		}
+		_, err = modifiedUserdataFile.Write(customScriptContent)
```

Comment:
If the custom script can get complex, would it benefit us to gzip encode it same as https://github.com/rancher/machine/blob/24ffad0c7eb9304db2a883664ab155e7dabb0344/commands/create.go#L650?